doc/migration.md

```diff
@@ -0,0 +1,424 @@
+
+| `sf::Vector3<T>` Function | Description                           |
+| ------------------------- | ------------------------------------- |
+| `length()`                | Get length                            |
+| `lengthSq()`              | Get length squared                    |
+| `normalized()`            | Get vector normalized to unit circle  |
+| `dot(sf::Vector2)`        | Get dot product                       |
+| `cross(sf::Vector2)`      | Get cross product                     |
+| `cwiseMul(sf::Vector2)`   | Perform component-wise multiplication |
+| `cwiseDiv(sf::Vector2)`   | Perform component-wise division       |
+
+## Threading Primitives
+
+`sf::Lock`, `sf::Mutex`, `sf::Thread`, `sf::ThreadLocal`, and `sf::ThreadLocalPtr` were removed and replaced with their equivalents from the standard library.
+The standard library provides multiple options for threads, locks, and mutexes among other threading primitives.
+
+| v2                   | v3                                               |
+| -------------------- | ------------------------------------------------ |
+| `sf::Lock `          | `std::lock_guard` or `std::unique_lock`          |
+| `sf::Mutex`          | `std::mutex`                                     |
+| `sf::Thread`         | `std::thread` or `std::jthread` (requires C++20) |
+| `sf::ThreadLocal`    | `thread_local`                                   |
+| `sf::ThreadLocalPtr` | `thread_local`                                   |
+
+## Deprecated APIs
+
+SFML 3 removes all of the deprecated APIs in SFML 2.
+
+| Deprecated API                      | Replacement                        |
+| ----------------------------------- | ---------------------------------- |
+| `sf::Event::MouseWheelEvent`        | `sf::Event::MouseWheelScrolled`    |
+| `sf::RenderWindow::capture`         | See 1                              |
+| `sf::RenderTexture::create`         | `sf::RenderTexture::resize`        |
+| `sf::Shader::setParameter`          | `sf::Shader::setUniform`           |
+| `sf::Text::setColor`                | `sf::Text::setFillColor`           |
+| `sf::Text::getColor`                | `sf::Text::getFillColor`           |
+| `sf::PrimitiveType::LinesStrip`     | `sf::PrimitiveType::LineStrip`     |
+| `sf::PrimitiveType::TrianglesStrip` | `sf::PrimitiveType::TriangleStrip` |
+| `sf::PrimitiveType::TrianglesFan`   | `sf::PrimitiveType::TriangleFan`   |
+| `sf::PrimitiveType::Quads`          | See 2                              |
+
+1. `sf::RenderWindow::capture` can be recreated by using an `sf::Texture` and its `sf::Texture::update(const Window&)` function to copy its contents into an `sf::Image` instead.
+2. `sf::PrimitiveType::Quads` can be replaced by another primitive type.
+This is not a drop-in replacement but rather will require refactoring your code to work with a new geometry.
+One viable option is to use `sf::PrimitiveType::TriangleStrip` where two adjacent triangles join to form what was previous one quad.
```

Comment:
That works only if the quads were connected in a particular way. A more general replacement is having two triangles with type `Triangles` for each quad.